web_src/js/components/RepoBranchTagSelector.vue

```diff
@@ -266,27 +294,15 @@ export function initRepoBranchTagSelector(selector) {
       isViewTree: false,
 
       active: 0,
-
+      isLoading: false,
+      // This means whether branch list/tag list has initialized
+      hasListInitialized: {
+        'branches': false,
+        'tags': false,
+      },
```

Comment:
These lines should be moved into the `data()` function

Reply:
This component does not have data() function yet as mentioned by the comment 
``` // no `data()`, at the moment, the `data()` is provided by the init code, which is not ideal and should be fixed in the future```

Reply:
> which is not ideal and should be fixed in the future

The old code was written as that because it came from legacy code by a refactoring PR (by copying&pasting to keep the old logic)

Since this PR changes a lot, maybe it's a chance for improving.

While it's not a must and it's not a block in my opinion.

Reply:
I checked this place, and I will leave this for another PR because it might need some other tests that are out of the scope of this PR.